docs/architecture/blazor-for-web-forms-developers/migration.md

```diff
@@ -8,8 +8,322 @@ ms.date: 09/11/2019
+  will be discussed more in depth in the chapter on
+  [Middleware](./middleware.md).
+
+## Example of code changes
+
+We'll demonstrate some of these code changes by looking at the [eShopOnBlazor](https://github.com/dotnet-architecture/eShopOnBlazor) project. This project is a simple eShop that allows a user to perform simple CRUD operations on a catalog. We will look specifically at the Create page and the changes required to make that work.
+
+### Start up changes
+
+In Blazor, the startup process is the same as other ASP.NET Core frameworks.
+This is greatly simplified from traditional ASP.NET where everything that is
+included is in the start up class itself. Comparing `Global.asax.cs` vs
+`Startup.cs`, the separation of the start up into services and pipeline make it
+clear what is being registered. In WebForms, dependency injection was not
+something that was built into the project, but something you could add on. For
+projects that wanted to use it, it was difficult to get things hooked up right,
+as can be seen in `Global.asax.cs` as the container has to be wired up
+manually.This is done for you in ASP.NET Core. Anything that is required for
+startup that had been in `Global.asax` can be added into this file to configure
+the ASP.NET pipeline.
+
+### Code behind vs razor page
+
+In WebForms, a page was defined by three files: Create.aspx,
+Create.aspx.designer.cs and Create.aspx.cs. In Blazor, it is all done in the
+same file, with any code generation from the HTML elements occurring behind the
+scenes (equivalent to the `*designer.cs` files).
+
+The first thing that you will see is that the syntax is slightly different from
+WebForms to Razor. WebForms components are brought in by a tag similar to
+`<asp:SomeComponent>` while in Blazor this will look like `<SomeComponent>`. The
+Razor compiler will understand that this is a Razor component and will handle it
+appropriately. The syntax to start injecting code is different, too; instead of
+`<%= %>` or `<%# =>`, code can be injected into a razor file by adding `@`
+before.
+
+For example, in order to add a field to the Create component, we change the
+following WebForms:
+
+```
+<div class="form-group">
+    <label class="control-label col-md-2">Name</label>
+    <div class="col-md-3">
+        <asp:TextBox ID="Name" runat="server" CssClass="form-control"></asp:TextBox>
+        <asp:RequiredFieldValidator runat="server" ControlToValidate="Name" Display="Dynamic"
+            CssClass="field-validation-valid text-danger" ErrorMessage="The Name field is required." />
+    </div>
+</div>
+```
+
+to
+
+```
+<div class="form-group">
+    <label class="control-label col-md-2">Name</label>
+    <div class="col-md-3">
+        <InputText class="form-control" @bind-Value="_item.Name" />
+        <ValidationMessage For="(() => _item.Name)" />
+    </div>
+</div>
+```
+
+Notice how `<InputText>` and `<ValidationMessage>` look just like HTML
```

Comment:
We should have some text here talking about what the Web Forms controls in this example are doing and then describe the equivalent Blazor components or logic.

Reply:
I think we also want an entire section in this doc that covers the most commonly used web server controls built into Web Forms, and show what the equivalent Blazor component code would look like. We don't have to do all the controls - just the most common ones to help with the migration.